mood_bot.py

```diff
@@ -0,0 +1,726 @@
+        else:
+            if day_input.lower() == 'everyday':
+                for i in range(7):
+                    mood_days[i] = 1
+                break
+            elif day_input.lower() == 'weekdays' or day_input.lower() == 'weekday':
+                for j in range(5):
+                    mood_days[j] = 1
+            elif day_input.lower() == 'weekends' or day_input.lower() == 'weekend':
+                for k in range(5, 7):
+                    mood_days[k] = 1
+            else:
+                mood_days[day_dictionary[day_input.lower()]] = 1
+    schedule_filename = str(author) + "_mood_schedule"
+    schedule_file = shelve.open(schedule_filename)
+    schedule_file['days'] = mood_days
+    schedule_file.close()
+    await new_schedule_display_message(context, 'days', num_days_to_word_days(mood_days))
+
+
+@mood_bot.command(name='add_time', help='Add time(s) (HH:MM) to mood schedule.',
+                  aliases=['addtime', '+time', 'time_add', 'timeadd', 'add_times', 'addtimes', '+times'])
+async def add_time(context, *time_inputs):
+    """This function allows the user to add a time of day to their mood check schedule."""
+    author = context.author
+    [_, mood_times] = mood_schedule(author)
+    for time_input in time_inputs:
+        try:
+            datetime.datetime.strptime(time_input, '%H:%M')
+        except ValueError:
+            await time_format_error(context, time_input)
+        if time_input in mood_times:
+            await duplicate_time_error(context, time_input)
+        else:
+            mood_times.append(time_input)
+    times_as_datetimes = []
+    for HM_time in mood_times:
+        times_as_datetimes.append(datetime.datetime.strptime(HM_time, '%H:%M'))
+    times_as_datetimes.sort()
+    new_times = []
+    for time_as_datetime in times_as_datetimes:
+        new_times.append(datetime.datetime.strftime(time_as_datetime, '%H:%M'))
+    schedule_filename = str(author) + "_mood_schedule"
+    schedule_file = shelve.open(schedule_filename)
+    schedule_file['times'] = new_times
+    schedule_file.close()
+    await new_schedule_display_message(context, 'times', new_times)
+
+
+@mood_bot.command(name='print_schedule', help='Prints current mood check schedule.',
+                  aliases=['printschedule', 'print_sched', 'printsched'])
+async def print_schedule(context):
+    """This function prints out the mood check-in schedule for the user in the format:
+    days of the week at times of day."""
+    author = context.author
+    [days, times] = mood_schedule(author)
+    word_days = num_days_to_word_days(days)
+    await context.send("Your mood check schedule is " + str(word_days) + " at " + str(times))
+
+
+@mood_bot.command(name='delete_day', help='Delete day(s) from mood check schedule.',
+                  aliases=['del_day', 'delday', 'deleteday', '-day', 'del_days', 'delete_days',
+                           'deldays', 'deletedays', '-days'])
+async def delete_day(context, *day_inputs):

```

Comment:
We might want the day schedule to be a hash / dictionary instead of a list / array